app/controllers/comment_controller.rb

```diff
@@ -1,3 +1,16 @@
+class CommentError < ArgumentError
+end
+
+def create_comment(node, user, body)
```

Comment:
This type of logic should go in helpers. Keep the controller code as thin as possible. Only keep action related methods